### PR TITLE
InputStream concurrency issue - Reduction to exceptions

### PR DESCRIPTION
**Pull request by LDuncAndroid**
- Reducing number of exceptions that occur when attempting to request multiple commands from the OBD by blocking the calls to send command and waiting for response.

I am submitting this as a potential reduction to the number of exceptions occurring when trying to issue commands, similar to the issues raised in #14 and #7

### CODE REVIEW

File: src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt [hunk 1]
```diff
@@ -45,32 +45,37 @@ class ObdDeviceConnection(private val inputStream: InputStream, private val outp
...
-            if (b < 0) {
-                break
-            }
-            c = b.toChar()
-            if (c == '>') {
-                break
+        withContext(Dispatchers.IO) {
+            // read until '>' arrives OR end of stream reached (-1)
+            while (inputStream.available() > 0) {
+                //println("InputStream Available: ${inputStream.available()}")
```
Comment (eltonvs) [inline src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt:66]: Nit: Can we remove this commented-out line?

**Review by eltonvs — APPROVED** [review]
Hey, @LDuncAndroid thanks for contributing to the library! 😄 

Nice catches on the `suspend` usages, this was the v1 and I wasn't able to come back on this solution 😓

I left one minor nit re: a commented-out line, but other than this, LGTM!

File: src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt (hunk 1 above)
Comment (LDuncAndroid) [reply src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt:66]: No problem, will get it removed. Thank you for the review.

File: src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt [hunk 2]
```diff
@@ -36,41 +36,45 @@ class ObdDeviceConnection(private val inputStream: InputStream, private val outp
...
-            if (c == '>') {
-                break
+        withContext(Dispatchers.IO) {
+            // read until '>' arrives OR end of stream reached (-1)
+            while (inputStream.available() > 0) {
+                b = inputStream.read().toByte()
+                if (b < 0) {
+                    break
+                }
+                c = b.toChar()
```
Comment (eltonvs) [inline src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt:70]: Just one small thing 😅
Seems like in the newest kotlin version, we can't convert directly from bytes to char, having to convert to int first. Would you mind changing this line to this instead?
```suggestion
                c = b.toInt().toChar()
```

**Review by eltonvs — COMMENTED** [review]
Thank you very much for making the changes, I just left one extra small comment

**Comment by LDuncAndroid** [comment]
@eltonvs thank you for merging. Glad I could help.

**Pull request merged**